src/commands/react.ts

```diff
@@ -0,0 +1,168 @@
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
+    }
+
+    let fetchedMessage: Discord.Collection<string, Message> = await message.channel
+      .fetchMessages({limit: 1, before: message.id});
+    if (!fetchedMessage || !(fetchedMessage instanceof Discord.Collection) ||
+      !(fetchedMessage.array()) || fetchedMessage.array().length < 1) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find a message to react to!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not find target message', result: {success: false}};
+    }
+
+    const targetMessage = fetchedMessage.array()[0];
+
+    const success = await this.reactService.tempReactToMessage(targetMessage, emoji, message);
+
+    if (success === false) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not apply the reaction!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not apply the reaction', result: {success: false}};
+    }
+
+    return {};
+  }
+
+  /**
+   * React to a specific message given by its Id
+   * @param messageId - The Id of the specific message
+   * @param emojiName - The name of the emoji
+   * @param message - The command message
+   */
+  async reactToMessage(messageId: string, emojiName: string, message: Discord.Message): Promise<ICommandResult> {
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
+    }
+
+    let initialId = message.channel.id;
+    let channels: Channel[] = [];
+    channels.push(message.channel);
+
+    // Get all guild channels if in a guild chat
+    if (message.channel.type === 'text') {
+      let guild = message.guild;
+      guild.channels.forEach((guildChannel) => {
+        if (guildChannel.type === 'text' && guildChannel.id !== initialId) {
+          channels.push(guildChannel as TextChannel);
+        }
+      });
+    }
+
+    // Loop through channels to try to apply emoji
+    let success = false;
+    while (channels.length > 0) {
+      let channel = channels.shift();
+      let textBasedChannel = channel as TextChannel | DMChannel | GroupDMChannel;
+      if (!textBasedChannel || !textBasedChannel.lastMessageID) { continue; }
+      try {
+        let targetMessage = await textBasedChannel.fetchMessage(messageId);
```

Comment:
Use `const` if the variable isn't reassigned. This should be caught by the linter.

Reply:
- [ ] Use const if the variable isn't reassigned. This should be caught by the linter.

Reply:
We might need to update the linter, looks like this wasn't caught by it